src/Spectre.libClustering/Partition.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Partition.h
+* Provides class implementation that simplifies partition vectors.
+*
+Copyright 2017 Dariusz Kuchta
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Partition.h"
+
+namespace Spectre::libClustering
+{
+	Partition::Partition(gsl::span<unsigned int> partition)
+	{
+		std::map<unsigned int, int> labelsDictionary;
+
+		int currentIntLabel = 1;
+		for (auto label : partition)
+			if (labelsDictionary.find(label) == labelsDictionary.end())
+				labelsDictionary.emplace(label, currentIntLabel++);
+
+		m_Partition.reserve(partition.size());
+		for (int i = 0; i < partition.size(); i++)
+			m_Partition.push_back(labelsDictionary[partition[i]]);
```

Comment:
If the change proposed by Grzesiek is implemented, push_back needs to be replaces with traditional array access.